rules-testing/test/recipes-test.js

```diff
@@ -0,0 +1,236 @@
+const assert = require('assert');
+const firebase = require("@firebase/testing");
+// var expect = require('chai').expect;
+
+const MY_PROJECT_ID = "boq-devex-step-2020";
+
+const currentUser = {
+  uid: "user_erica",
+};
+
+const otherUser = {
+  uid: "user_not_erica",
+};
+
+const myAuth = {uid: currentUser.uid};
+
+let validRecipe = {
+  recipeName: "Vanilla Cake",
+  uploaderUid: currentUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: ["cake", "frosting", "sprinkles"],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],
+};
+
+const invalidRecipe = {
+  recipeName: "Unicorn Cake",
+};
+
+const invalidDataRecipe = {
+  recipeName: 4,
+  uploaderUid: currentUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: ["cake", "frosting", "sprinkles"],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],  
+};
+
+const noIngredientsRecipe = {
+  recipeName: "Galaxy Cake",
+  uploaderUid: currentUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: [],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],  
+};
+
+const otherUserRecipe = {
+  recipeName: "Vanilla Cake",
+  uploaderUid: otherUser.uid,
+  difficulty: "Beginner",
+  description: "This is a cake. I really want to eat it.",
+  ingredients: ["cake", "frosting", "sprinkles"],
+  tools: ["fork", "knife"],
+  images: ["default image"],
+  instructions: ["eat cake"],
+  extraInfo: "",
+  timestamp: 2,
+  ratings: [],
+  tags: ["dessert"],
+};
+
+const admin = getAdminFirestore();
+admin.settings({
+  host: "localhost:8000",
+  ssl: false,
+});
+
+
+function getFirestore(auth) {
+  return firebase.initializeTestApp({projectId: MY_PROJECT_ID, auth: auth}).firestore();
+}
+
+function getAdminFirestore() {
+  return firebase.initializeAdminApp({projectId: MY_PROJECT_ID}).firestore();
+}
+
+describe("Recipe Rules", () => {
+
+  it("Understands basic addition", () => {
```

Comment:
IDK how I feel about this specific test  :/  seems a little out of scope to be testing that typescript can add correctly here